app/src/components/pages/ProfessorsList.tsx

```diff
@@ -2,28 +2,52 @@ import React from 'react';
 import { DataGrid, GridColDef } from '@mui/x-data-grid';
 import { parseCalendarTeacher } from 'utils/utils';
 import classData from 'data/clean.json';
-import { Box } from '@mui/material';
+import { Box, ButtonBase } from '@mui/material';
 import SearchIcon from '@mui/icons-material/Search';
 import { Search, SearchIconWrapper, StyledInputBase } from 'components/styles/styles';
 import { useState } from 'react';
 import { ICalendarItem_Teacher } from 'components/shared/interfaces/timetable.interfaces';
 import Link from '@mui/material/Link';
-
-const columns: GridColDef[] = [
-  { field: 'id', headerName: 'ID', width: 70 },
-  { field: 'teacherName', headerName: 'Username', width: 130 },
-  {
-    field: 'link',
-    headerName: 'View Professor Profile',
-    width: 200,
-    renderCell: (params: any) => <Link href={params.value}> View profile </Link>
-  }
-];
+import { NavigateFunction, useNavigate } from 'react-router-dom';
 
 function ProfessorsList() {
   const allData: ICalendarItem_Teacher[] = parseCalendarTeacher(JSON.parse(JSON.stringify(classData)));
   const [search, setSearch] = useState<string>('');
   const [rows, setRows] = useState<ICalendarItem_Teacher[]>(allData);
+  const navigate: NavigateFunction = useNavigate();
+
+  const columns: GridColDef[] = [
+    { field: 'id', headerName: 'ID', width: 70 },
+    { field: 'teacherName', headerName: 'Username', width: 130 },
+    { field: 'faculty', headerName: 'Faculty', width: 100 },
+    {
+      field: 'link',
+      headerName: 'View Professor Profile',
+      width: 200,
+      renderCell: (params) => <Link href={params.value}> View profile </Link>

```

Comment:
I think we should be passing the # id not the username since thats what the backend expects. Will just need to change the buildup of this `.value` param

Reply:
done